### PR TITLE
feat(split-button, multi-action-button): add position prop

### LINKED ISSUE #6705: MultiActionButton dropdown is aligned to the right side of the button
### Description

When the dropdown is wider than the button itself, it is aligned to the right side of the button while the DS aligns to the left (it is also more natural). Note that align property of the component has no effect on that.

From the discussion:
"i'd recommend using left aligned (minor button example in your screenshot) for path to completion and legibility reasons. Only when there's not enough space (right side of screen) should we align right."

image

### Reproduction

https://stackblitz.com/edit/parsium-carbon-starter-msdduz?file=src%2FApp.tsx

### Steps to reproduce

_No response_

### JIRA ticket numbers (Sage only)

_No response_

### Suggested solution

_No response_

### Carbon version

133.0.5

### Design tokens version

_No response_

### Relevant browsers

Chrome

### Relevant OSs

MacOS

### Additional context

_No response_

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

### PR DESCRIPTION
**Pull request by nuria1110**
fix #6705

### Proposed behaviour

`SplitButton`:
- Add `position` prop allow consumers to change the side on which the menu is aligned to when opened. 

Screenshot 2024-05-08 at 12 32 17
Screenshot 2024-05-08 at 12 32 32

`MultiActionButton`:
- Popup menu is aligned to the left by default. 
- Add `position` prop allow consumers to change the side on which the menu is aligned to when opened. 

Screenshot 2024-05-08 at 12 32 47
Screenshot 2024-05-08 at 12 32 55

### Current behaviour

`SplitButton`:
- Component does not have prop to change the side on which the menu is aligned to when opened. 

`MultiActionButton`:
- Popup menu is aligned to the right by default.
- Component does not have prop to change the side on which the menu is aligned to when opened. 

### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
- [x] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Playwright automation tests added or updated if required
- [x] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [x] Tested in provided StackBlitz sandbox/Storybook
- [x] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### CODE REVIEW

File: src/components/split-button/split-button.component.tsx [hunk 1]
```diff
@@ -42,16 +42,19 @@ export interface SplitButtonProps
...
-  /** The text to be displayed in the SplitButton. */
+  /** The text to be displayed in the button. */
   text: string;
+  /** Sets rendering position of menu */
+  position?: "left" | "right";
 }
 
 export const SplitButton = ({
   align = "left",
+  position = "right",
```
[the remark this answers is not in the document]
Comment (nuria1110) [reply src/components/split-button/split-button.component.tsx:57]: Yes this is intentional as the default position on the designs is to the right for `SplitButton` and the left for `MultiActionButton`.

File: src/components/multi-action-button/multi-action-button.spec.tsx [hunk 2]
```diff
@@ -645,6 +646,21 @@ describe("MultiActionButton", () => {
...
+    it.each([
+      ["bottom-start", "left"],
+      ["bottom-end", "right"],
+    ])(
+      "should have placement set to %s when position prop is %s",
+      (placement, positionValue) => {
+        wrapper = render({ position: positionValue }, mount);
+        openAdditionalButtons(wrapper);
+
+        expect(wrapper.find(Popover).props().placement).toEqual(placement);
```
[the remark this answers is not in the document]
Comment (DipperTheDan) [reply src/components/multi-action-button/multi-action-button.spec.tsx:659]: To add to this, I'm happy for it to be ignored from the coverage as long as we have a PW test to cover it. When I eventually get around to sorting the Storybook Play Functions stuff, we'll also be able to capture this in a snapshot.

**Review by DipperTheDan — COMMENTED** [review]
Good work. Only thing I'd like to add is it may be nice to add the new `position` prop in to the Storybook Controls panel in the test stories for MultiActionButton and SplitButton.

**Review by DipperTheDan — DISMISSED** [review]

**Review by edleeks87 — DISMISSED** [review]

**Review by edleeks87 — DISMISSED** [review]

**Review by DipperTheDan — DISMISSED** [review]

**Review by divyajindel — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Pull request merged**